src/PrpShop/PRP/Surface/QMipmap.cpp

```diff
@@ -298,332 +322,141 @@ QMipmap::QMipmap(plCreatable* pCre, QWidget* parent)
-    tex->extractAlphaData(data, dds.fLinearSize);
-    dds.setData(dds.fLinearSize, data);
-    delete[] data;
-
-    dds.write(S);
-}
-
-static bool getJColorSurface(const plDDSurface& dds, plMipmap* tex)
-{
-    if ((dds.fFlags & plDDSurface::DDSD_HEIGHT) == 0 ||
-        (dds.fFlags & plDDSurface::DDSD_WIDTH) == 0 ||
-        (dds.fFlags & plDDSurface::DDSD_PIXELFORMAT) == 0) {
-        QMessageBox::critical(NULL, QObject::tr("Error importing JPEG"),
-                              QObject::tr("DDSurface does not contain required fields"),
-                              QMessageBox::Ok);
-        return false;
-    }
-    if ((dds.fPixelFormat.fFlags & plDDSurface::DDPF_RGB) == 0 ||
-        (dds.fPixelFormat.fBitDepth != 24) ||
-        (dds.fPixelFormat.fRBitMask != 0xFF0000) ||
-        (dds.fPixelFormat.fGBitMask != 0x00FF00) ||
-        (dds.fPixelFormat.fBBitMask != 0x0000FF)) {
-        QMessageBox::critical(NULL, QObject::tr("Error importing JPEG"),
-                              QObject::tr("DDS file should be in RGB888 format"));
-        return false;
-    }
+    QString file_ext = QFileInfo(filename).suffix();
+    if (file_ext == "dds") {
+        hsFileStream S;
+        if (!S.open(qstr2st(filename), fmCreate)) {
+            QMessageBox::critical(this, tr("Error exporting DDS"),
+                                  tr("Error: Could not open file %1 for writing").arg(filename),
+                                  QMessageBox::Ok);
+            return;
+        }
 
-    tex->Create(dds.fWidth, dds.fHeight, 0, plBitmap::kJPEGCompression, plBitmap::kRGB8888);
-    tex->setColorData(dds.getData(), dds.getDataSize());
-    return true;
-}
+        try {
+            plDDSurface dds;
+            dds.setFromMipmap(tex);
+            dds.write(&S);
 
-static bool getJAlphaSurface(const plDDSurface& dds, plMipmap* tex)
-{
-    if ((dds.fFlags & plDDSurface::DDSD_HEIGHT) == 0 ||
-        (dds.fFlags & plDDSurface::DDSD_WIDTH) == 0 ||
-        (dds.fFlags & plDDSurface::DDSD_PIXELFORMAT) == 0) {
-        QMessageBox::critical(NULL, QObject::tr("Error importing JPEG"),
-                              QObject::tr("DDSurface does not contain required fields"),
-                              QMessageBox::Ok);
-        return false;
-    }
-    if ((dds.fPixelFormat.fFlags & plDDSurface::DDPF_LUMINANCE) == 0 ||
-        (dds.fPixelFormat.fBitDepth != 8) ||
-        (dds.fPixelFormat.fLuminanceBitMask != 0xFF)) {
-        QMessageBox::critical(NULL, QObject::tr("Error importing JPEG"),
-                              QObject::tr("DDS file should be in Luminance 8-bit format"),
-                              QMessageBox::Ok);
-        return false;
+            S.close();
+        } catch (hsException& ex) {
+            QMessageBox::critical(this, tr("Error exporting DDS"),
+                                  QString::fromUtf8(ex.what()), QMessageBox::Ok);
+        }
     }
+    else if (file_ext == "jpg" || file_ext == "jpeg") {
```

Comment:
```suggestion
    } else if (file_ext == "jpg" || file_ext == "jpeg") {
```

Also, should probably handle `Jpg`, `JpEg`, `JPEG`, etc (as above)